pyop2/base.py

```diff
@@ -1579,6 +1579,9 @@ def copy(self, other, subset=None):
         :arg subset: A :class:`Subset` of elements to copy (optional)"""
         if other is self:
             return
+        if subset is None:
+            other.data[:] = self.data
```

Comment:
`self.data_ro`.

If you want to handle the subset case:
```
if subset is None:
    other.data[:] = self.data_ro
else:
    other.data[subset.indices] = self.data_ro[subset.indices]
```

Reply:
I think you need to assert that the datasets for `other` and `self` match (and that if there's a subset that it also matches).

Reply:
If we do this then can we delete `_copy_parloop`?